test/helpers/framework.go

```diff
@@ -120,12 +121,89 @@ func (tc *E2ETestFramework) CreateTestNamespace() string {
 		},
 	}
 	_, err := tc.KubeClient.CoreV1().Namespaces().Create(namespace)
-	if err != nil {
+	if err != nil && !errors.IsAlreadyExists(err) {
 		logger.Error(err)
 	}
 	return name
 }
 
+// forceCreate does client.Create(config), deleting an existing instance if necessary.
+// It adds the new object to the cleanup list with deleteOpts
+// Panics if the client or config values are invalid or mismatched.
+func (tc *E2ETestFramework) forceCreate(client, config interface{}, deleteOpts *metav1.DeleteOptions) (newConfig interface{}, err error) {
+	rclient := reflect.ValueOf(client)
+	createMethod := rclient.MethodByName("Create")
+	deleteMethod := rclient.MethodByName("Delete")
+	if !createMethod.IsValid() || !deleteMethod.IsValid() {
+		panic(fmt.Errorf("not a kube client interface: %#v", client))
+	}
+	name := reflect.ValueOf(config).Elem().FieldByName("Name")
+	if !name.IsValid() {
+		panic(fmt.Errorf("not a valid config object, no Name: %#v", config))
+	}
```

Comment:
Why do we use `interface{}` and reflection. The k8s client-go/api-machinery offers `runtime.object` for this.

Reply:
Was not aware, thanks! Not sure how it allows us to avoid reflection, can you provide code?
```
type Object interface {
    GetObjectKind() schema.ObjectKind
    DeepCopyObject() Object
}
```
That gives us a clean way to get Kind, but not Name. We also need a way to call the Create, Delete and Update methods on the client, which don't have the same signatures for different kinds of object.
I'm sure we can find a cleaner way to do this somewhere in the k8s libraries though.